lib/Dialect/MSFT/MSFTOpInterfaces.cpp

```diff
@@ -7,10 +7,28 @@
 //===----------------------------------------------------------------------===//
 
 #include "circt/Dialect/MSFT/MSFTOpInterfaces.h"
+#include "circt/Dialect/MSFT/MSFTOps.h"
 
 using namespace circt;
 using namespace msft;
 
+LogicalResult circt::msft::verifyDynInstData(Operation *op) {
+  auto didOp = dyn_cast<DynInstDataOpInterface>(op);
+  if (!didOp)
```

Comment:
Should this be an assert?

Reply:
I think this can be an assert. If something other than the DynInstDataOpInterface is calling this it's probably because of a compiler writer error. It should always be called on an operation that `isa<DynInstDataOpInterfce>`

Reply:
Or, make this a `cast` not a `dyn_cast`.